### PR TITLE
Suggestion storage to connect to PouchDB

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: runtime/suggestion-storage.js [hunk 1]
```diff
@@ -49,8 +49,17 @@ export class SuggestionStorage {
...
+    let storage = this._arc._storageProviderFactory._storageForKey(storageKey);
+    let promise;
+    // TODO: the _join method exists for firebase storage and either connects or constructs
+    // depending on whether the storage exists. For pouch, the connect method does it.
+    // Should have a unified interface for this.
+    if (storage._join) {
+      promise = storage._join(
         id, type, storageKey, /* shoudExist= */ 'unknown', /* referenceMode= */ false);
+    } else {
+      promise = storage.connect(id, type, storageKey);
```
Comment (shans) [inline runtime/suggestion-storage.js]: I think you should use construct for pouch, not connect?

Comment (lindner) [reply runtime/suggestion-storage.js]: I think so too.  construct uses mustExist == false

```
    return this._join(id, type, keyFragment, false, referenceMode);
```

and connect uses mustExist == true:

```
    return this._join(id, type, key, true);
```

'unknown' doesn't exactly map, but it's more like baseStorageFor usage:

```
    const storagePromise = this._join(type.toString(), type.collectionOf(), key, 'unknown') as Promise<FirebaseCollection>;
```

which hits this code:

```
    if (shouldExist !== 'unknown' && shouldExist !== currentSnapshot.exists()) {
      return null;
    }
```

so yes, construct feels more apropos

Comment (lindner) [reply runtime/suggestion-storage.js]: Perhaps call  construct and then

	  provider.referenceMode = false;

Comment (mmandlis) [reply runtime/suggestion-storage.js]: addressed comment (the code is kinda ugly at the moment, so the storage api update will be a blessing). PTAL

**Comment by mmandlis** [comment]
addressed as part of: https://github.com/PolymerLabs/arcs/pull/2128

**Pull request closed without merging**